src/store-modules/active-media-store.js

```diff
@@ -0,0 +1,39 @@
+import {
+  SET_ACTIVE_MEDIA_ITEM,
+  UNSET_ACTIVE_MEDIA_ITEM,
+} from './mutation-types'
+
+/**
+ * Stores information about the active media item.
+ * @type {import('./types').ActiveMediaState}
+ */
+const state = {
+  activeMediaType: null,
+  activeMediaId: null,
+}
+
+const mutations = {
+  /**
+   * Set the active media item.
+   * @param {import('./types').ActiveMediaState} _state
+   * @param {'image' | 'audio' | null} type - the nature of the active media item
+   * @param {string} id - the ID of the active media item
```

Comment:
For object destructuring to work with TypeScript compiler we'd need to do the following:

```suggestion
   * @param {import('./types').ActiveMediaState} _state
   * @param {object} payload
   * @param {'image' | 'audio' | null} payload.type - the nature of the active media item
   * @param {string} payload.id - the ID of the active media item
```

Reply:
Thank you! I was stupidly blaming WebStorm for not being able to infer the types 😂!